src/Core/Grid/Definition/Factory/AddressGridDefinitionFactory.php

```diff
@@ -0,0 +1,297 @@
+                    ->setTypeOptions([
+                        'attr' => [
+                            'placeholder' => $this->trans('Search ID', [], 'Admin.Actions'),
+                        ],
+                        'required' => false,
+                    ])
+                    ->setAssociatedColumn('id_address')
+            )
+            ->add(
+                (new Filter('firstname', TextType::class))
+                    ->setTypeOptions([
+                        'attr' => [
+                            'placeholder' => $this->trans('Search first name', [], 'Admin.Actions'),
+                        ],
+                        'required' => false,
+                    ])
+                    ->setAssociatedColumn('firstname')
+            )
+            ->add(
+                (new Filter('lastname', TextType::class))
+                    ->setTypeOptions([
+                        'attr' => [
+                            'placeholder' => $this->trans('Search last name', [], 'Admin.Actions'),
+                        ],
+                        'required' => false,
+                    ])
+                    ->setAssociatedColumn('lastname')
+            )
+            ->add(
+                (new Filter('address1', TextType::class))
+                    ->setTypeOptions([
+                        'attr' => [
+                            'placeholder' => $this->trans('Search address', [], 'Admin.Actions'),
+                        ],
+                        'required' => false,
+                    ])
+                    ->setAssociatedColumn('address1')
+            )
+            ->add(
+                (new Filter('postcode', TextType::class))
+                    ->setTypeOptions([
+                        'attr' => [
+                            'placeholder' => $this->trans('Search post code', [], 'Admin.Actions'),
+                        ],
+                        'required' => false,
+                    ])
+                    ->setAssociatedColumn('postcode')
+            )
+            ->add(
+                (new Filter('city', TextType::class))
+                    ->setTypeOptions([
+                        'attr' => [
+                            'placeholder' => $this->trans('Search city', [], 'Admin.Actions'),
+                        ],
+                        'required' => false,
+                    ])
+                    ->setAssociatedColumn('city')
+            )
+            ->add(
+                (new Filter('id_country', CountryChoiceType::class))
+                    ->setTypeOptions([
+                        'required' => false,
+                    ])
+                    ->setAssociatedColumn('country_name')
+            )
+            ->add(
+                (new Filter('actions', SearchAndResetType::class))
+                    ->setAssociatedColumn('actions')
+                    ->setTypeOptions([
+                        'reset_route' => 'admin_common_reset_search',
```

Comment:
Shouldn't you use the filterId feature rather, the filters matched by controller/action are a bit outdated since they have limitations (even though it works fine for One grid pages)

Reply:
Yeh it's done. Sometimes it becomes difficult to follow witch method is the most recent and the one to use

Reply:
Sure, there are many things to remember And we lack documentation as well I planned to add som about this filtering feature as soon as I have time ^^